crates/sargon/src/types/samples/hierarchical_deterministic_factor_instance_samples.rs

```diff
@@ -0,0 +1,118 @@
+use crate::prelude::*;
+
+impl HierarchicalDeterministicFactorInstance {
+    pub(crate) fn sample_id_to_instance(
+        entity_kind: CAP26EntityKind,
+        index: HDPathComponent,
+    ) -> impl Fn(FactorSourceIDFromHash) -> Self {
+        move |id: FactorSourceIDFromHash| {
+            Self::sample_mainnet_tx(entity_kind, index, id)
+        }
+    }
+
+    pub fn sample_mainnet_tx(
+        entity_kind: CAP26EntityKind,
+        index: HDPathComponent,
+        factor_source_id: FactorSourceIDFromHash,
+    ) -> Self {
+        let derivation_path: DerivationPath = match entity_kind {
```

Comment:
You should probably create a prod (not `cfg(test)` constructor on `DerivationPath` which does this, and use that here. That constructor might be useful elsewhere!